### PR TITLE
Link group ownership boxes through filtered catalog page

### PR DESCRIPTION
**Pull request by prasaria**
## Hey, I just made a Pull Request!

Pull Request for this issue : https://github.com/backstage/backstage/issues/6681

I want to implement an active link if one of these boxes (eg: Services) is clicked , the page will moved to a filtered entity catalog page that is belong to that particular group.

What does this PR change : 
- Dynamically show top 6 entity types in boxes (does not include 0 count)
- Make these ownership boxes become clickable, it **will open a new tab** to a filtered catalog page
- Remove tooltip logic

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

Screen Shot 2021-08-13 at 22 10 37
Screen Shot 2021-08-13 at 22 11 59
Screen Shot 2021-08-13 at 22 12 20
Screen Shot 2021-08-13 at 22 12 35

### CODE REVIEW

**Comment by prasaria** [comment]
I still have failing test for `OwnershipCard.test.tsx`, maybe can you help me on how to fix this ? 

```
    console.error
      Error: Uncaught [Error: Missing required config value at 'app.baseUrl']
```

**Comment by luxaritas** [comment]
It was surprising when I first encountered this that I couldn't click through to a list of the relevant components, so thank you for this!

I'm wondering, is it necessary to specifically pin it to the top 6 categories? Maybe that number should be configurable, or be able to expand to show additional boxes somehow? Or maybe even an option to show specific boxes even if they are 0 for consistency. Really though I guess what my main concern would be is not having a direct line from this page to all owned components - maybe a single link to the catalog page that filters just by owner would be sufficient to resolve that? Just spitballing here.

**Comment by freben** [comment]
A contribution for an `All owned entities` button/link might make sense separately.

**Comment by prasaria** [comment]
Hi @luxaritas , thanks for your comment. Yeah that will be a great idea too if we can show more entity boxes, I'm open to any suggestion though but for now we need to hear something back from the backstage team first. 

I think the changes already include all the things that @timbonicus asked  in my issue: https://github.com/backstage/backstage/issues/6681, unless there are still something that I missed in my code.

File: plugins/org/src/components/Cards/OwnershipCard/OwnershipCard.tsx [hunk 1]
```diff
@@ -75,67 +68,53 @@ const useStyles = makeStyles((theme: BackstageTheme) =>
     bold: {
       fontWeight: theme.typography.fontWeightBold,
     },
-    service: {
+    box1: {
       background: createPageTheme(theme, 'home', 'service'),
```
Comment (timbonicus) [inline plugins/org/src/components/Cards/OwnershipCard/OwnershipCard.tsx:72]: I think you might be able to use the actual types here; the declaration can be a function that accepts props:

```suggestion
      background: props => createPageTheme(theme, props.types.length > 1 ? props.types[1] : 'none'),
```

Then call useStyles with the list of types further down:
```
const classes = useStyles({ types: topN.map(c => c.type) });
```

File: plugins/org/src/components/Cards/OwnershipCard/OwnershipCard.tsx [hunk 2]
```diff
@@ -149,10 +128,24 @@ const EntityCountTile = ({
...
 
+const getFilteredUrl = (
+  owner: Entity,
+  type: string,
+  entityKind: string,
+  baseUrl: string,
+): string => {
+  const ownerName = owner.metadata.name;
+  const filteredUrl = `
+    ${baseUrl}/catalog/?filters[kind]=${entityKind}&filters[type][0]=${type}&filters[user]=all&filters[owners][0]=${ownerName}
```
[the remark this answers is not in the document]
Comment (prasaria) [reply plugins/org/src/components/Cards/OwnershipCard/OwnershipCard.tsx]: updated @timbonicus

File: plugins/org/src/components/Cards/OwnershipCard/OwnershipCard.tsx [hunk 3]
```diff
@@ -185,56 +180,52 @@ export const OwnershipCard = ({
...
+    }, {} as Record<string, number>);
+
+    // Get top 6 entity types to be displayed in OwnershipCard
+    const topSixEntityTypes = Object.entries(countByType)
+      .sort(([, count1], [, count2]) => count2 - count1)
+      .map(([type]) => type)
+      .slice(0, 6);
+
+    return topSixEntityTypes.map((entityType, index) => ({
+      counter: countEntitiesBy(
```
[the remark this answers is not in the document]
Comment (prasaria) [reply plugins/org/src/components/Cards/OwnershipCard/OwnershipCard.tsx]: updated @timbonicus

File: plugins/org/src/components/Cards/OwnershipCard/OwnershipCard.tsx (hunk 1 above)
Comment (prasaria) [reply plugins/org/src/components/Cards/OwnershipCard/OwnershipCard.tsx:80]: I still dont get what you mean, do we really need to solve this now ?

**Review by Rugvip — COMMENTED** [review]
Thanks! 👍

File: plugins/org/src/components/Cards/OwnershipCard/OwnershipCard.tsx (hunk 1 above)
Comment (prasaria) [reply plugins/org/src/components/Cards/OwnershipCard/OwnershipCard.tsx:80]: What if the entity type is not described in the `pageTheme`, how should I handle that ?  Currently backstage  have 9 themes ( 'home' , 'service', etc...) If it is not match with the available theme list, does assigning a default value a good solution ?

File: plugins/org/src/components/Cards/OwnershipCard/OwnershipCard.tsx [hunk 4]
```diff
@@ -149,10 +125,30 @@ const EntityCountTile = ({
...
+  const type = selectedEntity.type;
+  const queryParams = qs.stringify({
+    filters: {
+      kind,
+      type,
+      owners: ownerName,
+      user: 'all',
+    },
+  });
+  const filteredUrl = generatePath(`/catalog/?${queryParams}`);
```
[the remark this answers is not in the document]
Comment (timbonicus) [reply plugins/org/src/components/Cards/OwnershipCard/OwnershipCard.tsx:147]: Ah, even better - sorry to churn this with my suggestion 😅

[the remark this answers is not in the document]
Comment (prasaria) [reply plugins/org/src/components/Cards/OwnershipCard/OwnershipCard.tsx]: ok updated, I hope it is already correct 😅

File: .changeset/cool-dolphins-join.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-org': minor
```
[the remark this answers is not in the document]
Comment (prasaria) [reply .changeset/cool-dolphins-join.md]: updated

File: plugins/org/src/components/Cards/OwnershipCard/OwnershipCard.tsx (hunk 1 above)
Comment (prasaria) [reply plugins/org/src/components/Cards/OwnershipCard/OwnershipCard.tsx]: updated

**Review by timbonicus — APPROVED** [review]
This looks good to me 🐬  Thank you @prasaria!

I was a bit on the fence with the all-caps but I can see from the screenshot the difficulty of different casings (gRPC, OpenAPI, etc) that are delivered lowercase in the data here. I can't think of a good solution. I think this is reasonable given that MUI's buttons use all-caps.

**Comment by prasaria** [comment]
Thank you @timbonicus,  can u merge it also ?

**Pull request merged**